### PR TITLE
Improvements to the CLI commands

### PR DESCRIPTION
**Pull request by renatonascalves**
- Improving PHPDoc
- More CLI command examples
- Adding more standards from https://make.wordpress.org/cli/handbook/guides/commands-cookbook/
- Using `WP_CLI::log` instead of `WP_CLI::line`
- The `debug` command checks for the existence of a valid post and adds a warning for `SP_DEBUG` and `SAVEQUERIES`
- Use `WP_CLI\Utils\get_flag_value` to get flag values
- Use quote instead of double quotes

### CODE REVIEW

File: bin/wp-cli.php [hunk 1]
```diff
@@ -67,76 +108,56 @@ public function put_mapping() {
...
-				'month' => date( 'm', $to ),
-				'day'   => date( 'd', $to ),
-			);
-		}
-		return $args;
-	}
-
 	/**
-	 * Index the current site or individual posts in elasticsearch, optionally flushing any existing data and adding the document mapping.
+	 * Index the current site or individual posts in ElasticSearch, optionally flushing any existing data and adding the document mapping.
```
Comment (mboynes) [inline bin/wp-cli.php:132]:
```suggestion
	 * Index the current site or individual posts in Elasticsearch, optionally flushing any existing data and adding the document mapping.
```

File: bin/wp-cli.php [hunk 2]
```diff
@@ -276,35 +298,93 @@ public function index( $args, $assoc_args ) {
 		$this->finish( $timestamp_start );
 	}
 
-
 	/**
-	 * Index a single post in elasticsearch and output debugging information. You should enable SP_DEBUG and SAVEQUERIES before running this.
+	 * Index a single post in ElasticSearch and output debugging information. You should enable `SP_DEBUG` and `SAVEQUERIES` before running this.
```
Comment (mboynes) [inline bin/wp-cli.php:302]:
```suggestion
	 * Index a single post in Elasticsearch and output debugging information. You should enable `SP_DEBUG` and `SAVEQUERIES` before running this.
```

File: bin/wp-cli.php [hunk 3]
```diff
@@ -67,76 +108,56 @@ public function put_mapping() {
 		}
 	}
 
-
 	/**
 	 * Flush the current index. !!Warning!! This empties your elasticsearch index for the entire site.
```
Comment (mboynes) [inline bin/wp-cli.php:112]:
```suggestion
	 * Flush the current index. !!Warning!! This empties your Elasticsearch index for the entire site.
```

File: bin/wp-cli.php [hunk 4]
```diff
@@ -276,35 +298,93 @@ public function index( $args, $assoc_args ) {
...
+			WP_CLI::error( 'Invalid post ID.' );
+		}
+
+		if ( ! defined( 'SP_DEBUG') ) {
+			WP_CLI::error( 'SP_DEBUG should be enabled before running this.' );
+		}
+
+		if ( ! defined( 'SAVEQUERIES') ) {
+			WP_CLI::error( 'SAVEQUERIES should be enabled before running this.' );
+		}
```
Comment (mboynes) [inline bin/wp-cli.php:327]: ooc, did you happen to check if it's too late at this point to set those? would be great to not require it.

Comment (mboynes) [reply bin/wp-cli.php:327]: at the very least, I think `error()` is too strong since it will kill the script. a warning is good enough.

Comment (renatonascalves) [reply bin/wp-cli.php:327]: Cool, will update! I interpreted _should_ as a _must_.

**Review by mogmarsh — APPROVED** [review]
some good improvements 👍

**Pull request merged**